### PR TITLE
feat: add coordinate systems

### PR DESCRIPTION
**Pull request by Zehua-Zeng**

### CODE REVIEW

File: draco/asp/define.lp [hunk 1]
```diff
@@ -3,6 +3,9 @@
 % @definition(mark_type) Types of marks to encode data.
 domain((mark,type),(point;bar;line;area;text;tick;rect)).
 
+% @definition(coordinate_type) Types of coordinates.
+domain((coordinate,type),(cartesian;polar)).
```
Comment (domoritz) [inline draco/asp/define.lp:7]: Shouldn't this be

```suggestion
domain((view,coordinate_system),(cartesian;polar)).
```

**Comment by domoritz** [comment]
Looking at these two options (this one and #320) , I feel this approach here with adding a coordinate system is simpler and cleaner. What do you think? If you don't have objections, I think we should merge this one.

**Comment by Zehua-Zeng** [comment]
> @JunranY @Zehua-Zeng Looking at these two options (this one and #320) , I feel this approach here with adding a coordinate system is simpler and cleaner. What do you think? If you don't have objections, I think we should merge this one.

I agree. I already fixed the conflicts, and now it is ready to merge.

**Review by domoritz — APPROVED** [review]

File: docs/facts/examples.ipynb [hunk 2]
```diff
@@ -118,6 +119,7 @@
     "    {\n",
     "        \"view\": [\n",
     "            {\n",
+    "                \"coordinate\": \"cartesian\",\n",
```
Comment (domoritz) [inline docs/facts/examples.ipynb:122]: This should not be required to be set manually, no?

**Review by domoritz — APPROVED** [review]

File: docs/facts/examples.ipynb (hunk 2 above)
Comment (Zehua-Zeng) [reply docs/facts/examples.ipynb:122]: This is not required to be set manually. The recommender can pick based on soft constraints. I just wanna add to examples to clarify which coordinate system the chart is using.

Comment (domoritz) [reply docs/facts/examples.ipynb:122]: I would prefer to remove unnecessary details.

File: docs/facts/view.md [hunk 3]
```diff
@@ -6,14 +6,16 @@ If a view has multiple marks, Draco assumes that the marks are layered (i.e. the
 
 ## View Properties
 
-By default, views don't have any properties.
+`(view,coordinate)`
```
Comment (domoritz) [inline docs/facts/view.md:9]:
```suggestion
`(view,coordinates)`
```

or

```suggestion
`(view,coordinate_system)`
```

**Review by domoritz — APPROVED** [review]
Thank you. Looks great with one naming change.

File: docs/facts/view.md (hunk 3 above)
Comment (Zehua-Zeng) [reply docs/facts/view.md]: Done

**Pull request merged**